### PR TITLE
Tweak env vars for local-up-cluster for CI

### PR DESCRIPTION
**Pull request by dims**
We need a few more things setup before local-up-cluster can work
well in the CI environment.

1. specify KUBERNETES_PROVIDER as local to pick up correct scripts
2. specify KUBERNETES_CONFORMANCE_TEST since we are running the
   conformance scenario. Picks up the host ip address from kubeconfig
3. Use the ip address of the docker0 network interface and use that
   for all the HOST/IP(s) that need to be configured.

### CODE REVIEW

**Comment by dims** [comment]
/hold cancel

tested it locally with the `gcr.io/k8s-testimages/kubekins-e2e:v20180223-74690e8dc-master` image and a locally built `kubetest`

File: kubetest/local.go [hunk 1]
```diff
@@ -70,10 +71,23 @@ func (n localCluster) Up() error {
 	if err != nil {
 		return err
 	}
+
 	cmd := exec.Command(script)
 	cmd.Env = os.Environ()
 	cmd.Env = append(cmd.Env, "ENABLE_DAEMON=true")
 	cmd.Env = append(cmd.Env, fmt.Sprintf("LOG_DIR=%s", n.tempDir))
+
+	getIp := `ip route list | grep docker0 | awk '{print $9}'`
```
Comment (krzyzacy) [inline kubetest/local.go]: comment what this does?

Comment (dims) [reply kubetest/local.go]: done thanks!

File: kubetest/local.go [hunk 2]
```diff
@@ -70,10 +71,27 @@ func (n localCluster) Up() error {
...
 	cmd := exec.Command(script)
 	cmd.Env = os.Environ()
 	cmd.Env = append(cmd.Env, "ENABLE_DAEMON=true")
 	cmd.Env = append(cmd.Env, fmt.Sprintf("LOG_DIR=%s", n.tempDir))
+
+	// when we are running in a DIND scenario, we should use the ip address of
+	// the docker0 network interface, This ensures that when the pods come up
+	// the health checks (for example for kubedns) succeed. If there is no
+	// docker0, just use the defaults in local-up-cluster.sh
+	getIp := `ip route list | grep docker0 | awk '{print $9}'`
```
Comment (krzyzacy) [inline kubetest/local.go]: I think you can do it in go by iterating `net.Interfaces()`?

like https://stackoverflow.com/questions/23558425/how-do-i-get-the-local-ip-address-in-go?

*bash is evil...*

Comment (dims) [reply kubetest/local.go]: yep, done! thanks for the nudge :)

Comment (krzyzacy) [reply kubetest/local.go]: (forgot to push?)

Comment (dims) [reply kubetest/local.go]: whoops!. pushed

File: kubetest/local.go [hunk 3]
```diff
@@ -70,10 +71,39 @@ func (n localCluster) Up() error {
...
 	cmd.Env = append(cmd.Env, "ENABLE_DAEMON=true")
 	cmd.Env = append(cmd.Env, fmt.Sprintf("LOG_DIR=%s", n.tempDir))
+
+	// when we are running in a DIND scenario, we should use the ip address of
+	// the docker0 network interface, This ensures that when the pods come up
+	// the health checks (for example for kubedns) succeed. If there is no
+	// docker0, just use the defaults in local-up-cluster.sh
+	dockerIp := ""
+	docker0, err := net.InterfaceByName("docker0")
+	if err == nil {
```
Comment (krzyzacy) [inline kubetest/local.go]: bikeshed: maybe better to handle log errors here
```
if docker0, err := net.InterfaceByName("docker0"); err != nil {
  if addresses, err := docker0.Addrs(); err != nil {
    for _, address := range addresses {
      ...
    }
  } else {
    log.printf("error from docker0.Addrs() : %v", err)
  }
} else {
  log.printf("error from net.InterfaceByName : %v", err)
}

 I think, and you can print the error

Comment (krzyzacy) [reply kubetest/local.go]: or, probably should return the err?

Comment (dims) [reply kubetest/local.go]: @krzyzacy i'll just do the logging and fall back to the defaults in local-up-cluster.sh if we are unable to get info about docker0 net interface

Comment (krzyzacy) [reply kubetest/local.go]: sgtm

Comment (dims) [reply kubetest/local.go]: thanks updated.

**Pull request merged**